### PR TITLE
Migrate the ios_app_with_extensions to null safety

### PR DESCRIPTION
**Pull request by Abhishek01039**
I have migrated `dev/integration_tests/ios_app_with_extensions` with null safety.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Abhishek01039** [comment]
Hello @jmagman 
This example is not migrated to null safety
so when I am trying to migrate it
CI is not happy!

**Comment by Abhishek01039** [comment]
Hello @jmagman 
I am trying to migrate this example to null safety
but CI seems unhappy!
can you please take a look?
Thanks

**Comment by xu-baolin** [comment]
Try rebasing off the latest master and see if that fixes the tests.

**Comment by Abhishek01039** [comment]
Hello @xu-baolin 
I have rebased but one test failed.

**Comment by xu-baolin** [comment]
I reran it and passed now.

**Comment by Abhishek01039** [comment]
Thanks @xu-baolin

**Comment by Abhishek01039** [comment]
Hello @Hixie 
can you please mark it as test-exempt?
Thanks

File: dev/integration_tests/ios_app_with_extensions/pubspec.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ name: ios_app_with_extensions
 version: 1.0.0+1
 
 environment:
-  sdk: ">=2.1.0 <3.0.0"
+  sdk: ">=2.12.1"
```
Comment (jmagman) [inline dev/integration_tests/ios_app_with_extensions/pubspec.yaml:16]: Let's use the recommended version constraint https://dart.dev/null-safety#constraints
```suggestion
  sdk: ">=2.12.0 <3.0.0"
```

**Review by jmagman — COMMENTED** [review]
Ran this locally, `ios_app_with_extensions_test` passes.  No dart analyzer issues.

LGTM with sdk range nit.

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic changes

File: dev/integration_tests/ios_app_with_extensions/pubspec.yaml (hunk 1 above)
Comment (Abhishek01039) [reply dev/integration_tests/ios_app_with_extensions/pubspec.yaml]: Thanks @jmagman

**Review by jmagman — APPROVED** [review]
LGTM!

**Pull request merged**